### PR TITLE
Allow conversion from `CompositeId` to an erased `ArcCache` record

### LINKED ISSUE #1576: Allow conversion from `CompositeId` to an `ArcCache` record
Given a `CompositeId`, we should be able to possibly resolve: 

- resource_uri
- label

`ArcCache::get_record(composite_id:&CompositeId) -> Option<impl EndpointName + Label >`

### PR DESCRIPTION
**Pull request by jgrund**
Fixes #1576.

COPR Module Manager: managerforlustre/content-type-arc

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1580)

### CODE REVIEW

File: iml-warp-drive/src/cache.rs [hunk 1]
```diff
@@ -147,6 +147,12 @@ pub async fn db_record_to_change_record(
                 Ok(RecordChange::Update(Record::LnetConfiguration(x)))
             }
         },
+        DbRecord::ContentType(x) => match (msg_type, x) {
+            (MessageType::Delete, x) => Ok(RecordChange::Delete(RecordId::ContentType(x.id()))),
+            (MessageType::Insert, x) | (MessageType::Update, x) => {
+                Ok(RecordChange::Update(Record::ContentType(x)))
+            }
+        },
```
Comment (ip1981) [inline iml-warp-drive/src/cache.rs]: Looks like a part of https://github.com/whamcloud/integrated-manager-for-lustre/pull/1574?

**Comment by jgrund** [comment]
___
*[iml-wire-types/src/lib.rs, line 295 at r1](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1580#-M02j_DR200-Oj27hXaw:-M0303V64C7Xay45O-Lx:b-ccnzax) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/83c26b82c3fd26131e075bba5f313697c24e4cb6/iml-wire-types/src/lib.rs#L295)):*
<details><summary><i>Previously, nlinker (Nick Linker) wrote…</i></summary><blockquote>

Hm, what caused this constraint?
</blockquote></details>

All traits must be sized when used in a blanket implementation here:

https://github.com/whamcloud/integrated-manager-for-lustre/pull/1580/files#diff-2ab57dc4ef74128b853414471785ead6R184-R188

Since the `EndpointName` doesn't take self, it was unsized, hence the added bound.

**Comment by jgrund** [comment]
___
*[iml-wire-types/src/warp_drive.rs, line 261 at r1](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1580#-M02k4AGC4DQZ_Sk6Qaj:-M030ixAC2q6uoVjxUba:bv4humh) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/83c26b82c3fd26131e075bba5f313697c24e4cb6/iml-wire-types/src/warp_drive.rs#L261)):*
<details><summary><i>Previously, nlinker (Nick Linker) wrote…</i></summary><blockquote>

Where did these constants come from?
</blockquote></details>

They are in the model column of the django_content_types table. Not pretty, but needed a way to resolve from content_type_id -> ErasedRecord, and that translation needs to be known somewhere.

File: iml-warp-drive/src/cache.rs (hunk 1 above)
Comment (jgrund) [reply iml-warp-drive/src/cache.rs]: Yeah, I need to build out a patch stack until this and other work lands. I've updated the base branch of #1574 to point to this patch so those files will fall out of that PR.

**Pull request merged**